client/my-sites/plugins/categories/use-get-category-url.tsx

```diff
@@ -0,0 +1,18 @@
+import { useSelector } from 'react-redux';
+import { useLocalizedPlugins } from 'calypso/my-sites/plugins/utils';
+import { getSiteDomain } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+export function useGetCategoryUrl() {
+	const siteId = useSelector( getSelectedSiteId ) as number;
+	const domain = useSelector( ( state ) => getSiteDomain( state, siteId ) ) || '';
```

Comment:
I now realize that the subdirectory site sites are not supported by the category links.
For example: while on `/plugins/multisite.com::testSited1` clicking the SEO category will redirect to `/plugins/browse/seo/multisite.com/testSited1` because `getSiteDomain` returns the publicly accessible URL for that subdirectory site. The endresult is a blank page.

While we're at it, I think it makes sense to fix it here. Using `domain = domain.replace( /\//, '::' )` seems to work, but I'm open to better solutions.

Reply:
@claudiucelfilip I'm okay with the simple substitution. Also found [similar](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts#L779) [approaches](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/composite-checkout/lib/pending-page.ts#L233) in the Calypso codebase.